components/org.apache.stratos.rest.endpoint/src/main/java/org/apache/stratos/rest/endpoint/api/StratosApiV41.java

```diff
@@ -453,6 +453,18 @@ public Response addServiceGroup(
             if (e.getCause().getMessage().contains("already exists")) {
                 return Response.status(Response.Status.CONFLICT).entity(new StatusResponseBean(
                         Response.Status.CONFLICT.getStatusCode(), "Cartridge group not found")).build();
+            } else if (e.getCause().getMessage().contains("duplicate cartridges")) {
```

Comment:
IMO it would be better to use specific exceptions for handling this logic rather than using different messages.
Thanks